app/modules/enduser_setup.c

```diff
@@ -626,6 +626,52 @@ static void do_station_cfg (task_param_t param, uint8_t prio)
 }
 
 
+/**
+ *
+ *
+ *
+ *
+ */
+static int enduser_setup_write_file_with_extra_configuration_data(
+char * configuration_string, int configuration_length
+)
+{
+  int p_file = 0;
+  lua_State *L = NULL;
+	
+  ENDUSER_SETUP_DEBUG("enduser: opening enduser_custom_parameters.json for write");
+
+  // setup the file output
+  p_file = vfs_open("enduser_custom_parameters.json", "w");
+  if (p_file == 0)
+  {
+    ENDUSER_SETUP_DEBUG("Can't write to file!");
+    return 1;
+  }
+  
+  /* Not certain what this does */
+  L = lua_getstate();
+  
+  if (L != NULL) 
+  {
+    lua_pushlstring(L, configuration_string, configuration_length);
```

Comment:
Without the `lua_call()` below, this will end up leaving the stack in an inconsistent state. I'd suggest moving this down to the same block, there's no reason not to from what I can see.